### PR TITLE
Remove rhel6 template

### PR DESCRIPTION
**Pull request by fossedihelm**
Signed-off-by: fossedihelm <ffossemo@redhat.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Removed RHEL6 from the available templates
```

### CODE REVIEW

**Comment by fossedihelm** [comment]
/cc @ksimon1
/hold
Need to delete rhel6 job first

**Comment by fossedihelm** [comment]
Pushed https://github.com/openshift/release/pull/34004

**Comment by 0xFelix** [comment]
Looks good to me but notice that the CI is broken currently. Tests need to be retriggered once CI is fixed.

**Comment by ksimon1** [comment]
/retest
@geetikakay @dominikholler are you ok, with deleting this template?

**Comment by geetikakay** [comment]
Check : https://access.redhat.com/support/policy/updates/errata
for rhel6 i see Extended life cycle support (ELS) add-on ends : June 30, 2024.

**Comment by fossedihelm** [comment]
> Check : https://access.redhat.com/support/policy/updates/errata for rhel6 i see Extended life cycle support (ELS) add-on ends : June 30, 2024. Should we wait till then?

IMHO we should remove it before then. This is meant to be in 4.13, so that when rhel6 ELS ends, users are not able to run them. Thanks

**Comment by ksimon1** [comment]
@fossedihelm can you please fix failing unit test?

**Comment by fossedihelm** [comment]
> `Error: No default variant set for os.template.kubevirt.io/rhel6.1 = true`
> 
> @fossedihelm Can you take a look at this please?

@0xFelix PTAL! Thanks

File: automation/test_defaults.sh [hunk 1]
```diff
@@ -33,7 +33,7 @@ readarray -t oss < <( \
 for os in "${oss[@]}"; do
   defaults=$(oc get template -l "$os,template.kubevirt.io/default-os-variant = true,$ver_label" -o name | wc -l)
 
-  if [[ $defaults -eq 1 ]]; then
+  if [[ $defaults -eq 1 || "$os" =~ (^os.template.kubevirt.io/rhel6.* '=' true) ]]; then
```
Comment (0xFelix) [inline automation/test_defaults.sh:36]: This does not look right. Without any rhel6 templates, there shouldn't be any `os.template.kubevirt.io/rhel6.*` labels.

**Review by 0xFelix — COMMENTED** [review]
I just noticed that centos6 should be removed too, if we are dropping rhel6. Can you do that as well?

**Comment by ksimon1** [comment]
The failing unit test is coming from upgrade path: https://github.com/kubevirt/common-templates/blob/master/automation/unit-tests.sh#L50. After creation of previous released templates, remove all rhel 6 templates

File: automation/test_defaults.sh (hunk 1 above)
Comment (fossedihelm) [reply automation/test_defaults.sh]: Recap of offline talk: We decided to exclude from the defaullt check the templates which are not part of the current build. Thanks!

**Comment by fossedihelm** [comment]
> The failing unit test is coming from upgrade path: https://github.com/kubevirt/common-templates/blob/master/automation/unit-tests.sh#L50. After creation of previous released templates, remove all rhel 6 templates

Thanks! I talked to @0xFelix and we decided to take a more generic approach!
We decided to exclude from the `test_defaults` check the templates which are not part of the current build.
I made a mistake here:https://github.com/kubevirt/common-templates/commit/1e0a50cbd212d0fc8a5de9267b28b8a418806296 since I was not resetting the `found` variable at the beginning of each cycle.
I pushed a fix right now.
@ksimon1 Is it something that could work for you? Thanks

**Comment by ksimon1** [comment]
> > The failing unit test is coming from upgrade path: https://github.com/kubevirt/common-templates/blob/master/automation/unit-tests.sh#L50. After creation of previous released templates, remove all rhel 6 templates
> 
> Thanks! I talked to @0xFelix and we decided to take a more generic approach! We decided to exclude from the `test_defaults` check the templates which are not part of the current build. I made a mistake here:[1e0a50c](https://github.com/kubevirt/common-templates/commit/1e0a50cbd212d0fc8a5de9267b28b8a418806296) since I was not resetting the `found` variable at the beginning of each cycle. I pushed a fix right now. @ksimon1 Is it something that could work for you? Thanks

Yes, I am ok with that

**Review by 0xFelix — COMMENTED** [review]
What about something like this to get the deduplicated list of labels:

```shell
# Create an associative array containing all the possible labels
dist_templates=(dist/templates/*.yaml)
declare -A dist_templates_labels
for template in "${dist_templates[@]}"; do
  while read -r label; do
    dist_templates_labels[${label}]=""
  done < <( \
      yq -o props '.metadata.labels | with_entries(select(.key == "os.template.kubevirt.io/*"))' "$template" \
  )
done
```

And this to check for presence of a label?

```shell
if [[ ! -v dist_templates_labels[$os] ]]; then
  continue
fi
```

**Comment by fossedihelm** [comment]
> What about something like this to get the deduplicated list of labels:
> 
> ```shell
> # Create an associative array containing all the possible labels
> dist_templates=(dist/templates/*.yaml)
> declare -A dist_templates_labels
> for template in "${dist_templates[@]}"; do
>   while read -r label; do
>     dist_templates_labels[${label}]=""
>   done < <( \
>       yq -o props '.metadata.labels | with_entries(select(.key == "os.template.kubevirt.io/*"))' "$template" \
>   )
> done
> ```
> 
> And this to check for presence of a label?
> 
> ```shell
> if [[ ! -v dist_templates_labels[$os] ]]; then
>   continue
> fi
> ```

Definitely more beautiful and functional than my proposal! :) Thanks

**Comment by ksimon1** [comment]
@fossedihelm can you remove hold label?

**Comment by fossedihelm** [comment]
@ksimon1 I am not sure why `ci/prow/e2e-gcp-common-templates-rhel6` job succeed. I would expect it fails due the template does not exist anymore. Any idea? Thanks

**Comment by 0xFelix** [comment]
```
if [[ $TARGET =~ rhel6.* ]]; then
  workloads=("desktop" "server")
fi
```

By removing this `ci/prow/e2e-gcp-common-templates-rhel6` likely became an empty test job. Nothing to test -> nothing fails.

**Comment by fossedihelm** [comment]
> ```
> if [[ $TARGET =~ rhel6.* ]]; then
>   workloads=("desktop" "server")
> fi
> ```
> 
> By removing this, `ci/prow/e2e-gcp-common-templates-rhel6` likely became an empty test job. Nothing to test -> nothing fails.

:sweat_smile: trivially. Thanks

**Pull request merged**